core/solver/lower_trs.cpp

```diff
@@ -61,6 +62,38 @@ GKO_REGISTER_OPERATION(solve, lower_trs::solve);
 }  // namespace lower_trs
 
 
+template <typename ValueType, typename IndexType>
+std::unique_ptr<LinOp> LowerTrs<ValueType, IndexType>::transpose() const
+{
+    std::unique_ptr<transposed_type> transposed{
+        new transposed_type{this->get_executor()}};
+    transposed->set_size(gko::transpose(this->get_size()));
+    transposed->system_matrix_ = share(as<matrix::Csr<ValueType, IndexType>>(
+        this->get_system_matrix()->transpose()));
+    transposed->parameters_.num_rhs = this->parameters_.num_rhs;
+    transposed->init_trs_solve_struct();
+    transposed->generate();
+
```

Comment:
Why to use factory build in the CG/FCG/etc... but create empty, then change the parameter in the LowerTrs?